lib/ssz_types/beacon_chain/beacon_state.ex

```diff
@@ -89,16 +89,107 @@ defmodule SszTypes.BeaconState do
           historical_summaries: list(SszTypes.HistoricalSummary.t())
         }
 
+  alias LambdaEthereumConsensus.StateTransition.Accessors
+  alias LambdaEthereumConsensus.StateTransition.Predicates
+
   @doc """
     Decrease the validator balance at index ``index`` by ``delta``, with underflow protection.
   """
   @spec decrease_balance(t(), SszTypes.validator_index(), SszTypes.gwei()) :: t()
-  def decrease_balance(%{balances: balances} = state, index, delta) do
+  def decrease_balance(%__MODULE__{balances: balances} = state, index, delta) do
     current_balance = Enum.fetch!(balances, index)
 
     %{
       state
       | balances: List.replace_at(balances, index, max(current_balance - delta, 0))
     }
   end
+
+  @doc """
+  Return the deltas for a given ``flag_index`` by scanning through the participation flags.
+  """
+  @spec get_flag_index_deltas(t(), integer) :: {list(SszTypes.gwei()), list(SszTypes.gwei())}
```

Comment:
hmm, I feel this file will get quite bloated if we add all these helpers

Reply:
Yes, but this should be the last one